Objects/typeobject.c

```diff
@@ -3196,6 +3194,22 @@ _PyType_GetModuleByDef(PyTypeObject *type, struct PyModuleDef *def)
 }
 
 
+void
+PyType_SetBase(PyTypeObject *type, PyTypeObject *base)
+{
+    _PyObject_ASSERT((PyObject *)type, (type->tp_flags & Py_TPFLAGS_HEAPTYPE));
+    Py_INCREF(base);
+    Py_XSETREF(type->tp_base, base);
```

Comment:
Remove `Py_INCREF(base);` and
```suggestion
    Py_XSETREF(type->tp_base, Py_NewRef(base));
```